### ISSUE TITLE
Issue with ordering of "clazz" and "data" keys in Json message 

### ISSUE THREAD

**Issue by prasadtiru** [org-arl/fjage#38]
Tested on the dev branch. Following is the issue:

Two identical Json messages are generated, the first one is using python gateway and the second one is using `rconsole.sh` script. The json messages are following:

```
{  
   "action":"send",
   "message":{  
      "data":{  
         "cmd":"ps",
         "perf":"REQUEST",
         "msgID":"97ae40c2-4b3f-4ed7-ab12-42dcacbfdf71",
         "recipient":"shell",
         "sender":"PythonGW"
      },
      "clazz":"org.arl.fjage.shell.ShellExecReq"
   },
   "relay":true
}

```
```
{  
   "action":"send",
   "message":{  
      "clazz":"org.arl.fjage.shell.ShellExecReq",
      "data":{  
         "cmd":"ps",
         "msgID":"d189c247-952d-448c-8c3a-69ab2456c41b",
         "perf":"REQUEST",
         "recipient":"shell",
         "sender":"rshell"
      }
   },
   "relay":true
}

```
The only difference is the order of `"clazz"` and `"data"` keys. The issue is that `ShellExecReq` message is handled only when the `"clazz"` key appears before `"data"` key in the Json message. Now, since in python dictionary the keys are unordered, the json message built and sent from python gateway doesn't get expected response when the `"data"` key comes in before`"clazz"` key.

**Comment by mchitre** [comment]
Yes, both ``MessageAdapterFactory`` and ``GenericValueAdaptorFactory`` assume that the ``clazz`` key is read before the actual object. This is necessary to ensure that the JSON doesn't have to be parsed twice, and so is a performance enhancement, while it breaks strict JSON as it requires ordering of keys. The only workaround I can think of would have a large performance impact, since serialization/deserialization happens often. Can't you massage your Python JSON generation to create the JSON message always with ``clazz`` key as the first entry?

**Comment by prasadtiru** [comment]
Yes. Can handle that in Python. Will make sure `"clazz"` comes first.

**Comment by mchitre** [comment]
We'll keep issue open, to eventually be fully JSON compliant in fjåge, once we have a good way of handling it with minimal performance impact.
